R/coverage-matrix.R

```diff
@@ -0,0 +1,156 @@
+#' Scrape "coverage" tag in test code to generate mapping
+#' @param references dynamic reference holder if it already exists
+#' @param src,ref passed to \code{\link{vt_scrape_tags_from}}
+#' @param type one of "long" or "wide" which determines shape of output table
+#' @return a data.frame mapping requirement ids to test case ids.
+#' @importFrom rlang list2 := !!
+#' @export
+vt_scrape_coverage_matrix <- function(type = c("long", "wide"), references = NULL, src = ".", ref = vt_path()){
+
+  ## instantiate dynamic referencing
+  ## if no dynamic reference content, this won't get populated
+  if(is.null(references)){
+    references <- vt_dynamic_referencer$new()
+  }
```

Comment:
This would create a new referencer. by default if referencer is null in dynamic_reference_rendering, it will select the shared dynamic referencer from valtools.

Reply:
oh, because it requires a filename. 🤦 Lets maybe use that same paradigm though, rather than creating a new referencer

Reply:
Can we update `dynamic_reference_rendering` to handle R objects?

Reply:
Yupp! I think this is available and was merged in